test/file.ts

```diff
@@ -654,6 +653,78 @@ describe('File', () => {
     });
   });
 
+  describe('.copyRecursively', () => {
+    let shallowCopySource;
+    let deepCopySource;
+    let destinationPath;
+
+    const file = getFixturePath('shallowCopySource/a.txt');
+
+    beforeEach(() => {
+      fs.mkdirSync(getFixturePath('shallowCopySource'));
+      createFile(file, { duration: 1, modifier: 'hours' });
+      shallowCopySource = new File(getFixturePath('shallowCopySource/'));
+      destinationPath = getFixturePath('copyDest/');
+    });
+
+    afterEach(async () => {
+      await shallowCopySource.deleteRecursively();
+      const destination = new File(destinationPath);
+      await destination.deleteRecursively();
+    });
+
+    it('copies a directory and its contents to destination', async () => {
+      await shallowCopySource.copyRecursively(destinationPath);
+
+      assert.isTrue(fs.existsSync(destinationPath));
+      assert.exists(fs.statSync(`${destinationPath}a.txt`));
+    });
+
+    it('deep copies a directory and its sub-contents to destination', async () => {
+      fs.mkdirSync(getFixturePath('deepCopySource'));
+      createFileStructure('deepCopySource', 5);
+      deepCopySource = new File(getFixturePath('deepCopySource/'));
+
+      await deepCopySource.copyRecursively(destinationPath);
+      assert.isTrue(fs.existsSync(destinationPath));
+      assert.isTrue(fs.existsSync(`${destinationPath}subDir_0`));
+
+      await deepCopySource.deleteRecursively();
+    });
+
+    describe('overwrite', () => {
+      beforeEach(() => {
+        fs.mkdirSync(destinationPath);
+        const existingFile = getFixturePath('copyDest/originalFile.txt');
+        createFile(existingFile, { duration: 1, modifier: 'hours' });
+      });
+
+      context('true', () => {
+        it('overwrites existing destination when copying', async () => {
+          await shallowCopySource.copyRecursively(destinationPath, { overwrite: true });
+
+          assert.isTrue(fs.existsSync(destinationPath));
+          assert.exists(fs.statSync(`${destinationPath}a.txt`));
+          assert.isFalse(fs.existsSync(`${destinationPath}originalFile.txt`));
+        });
+      });
+
+      context('false', () => {
+        it('does not overwrite existing destination when copying', async () => {
+          try {
+            await shallowCopySource.copyRecursively(destinationPath);
+            throw new Error('Wrong error');
```

Comment:
An alternative is to return in the catch block and add an assert fail on the bottom on the function.